### PR TITLE
NO-JIRA Fix visual styles for white-background version

### PR DESCRIPTION
**Pull request by danlaush**
## Context

Visual bug spotted before go-live of the new Business nav

### Changes

On the non-inverse (white background) mobile view:
- Fix background colour of `Personal | Business` bar
- Fix text colour of dropdown links 

| Before ([View in browser](https://transferwise.github.io/public-navigation/)) | After ([View in browser](https://transferwise.github.io/public-navigation/branch/inverse-style-fixes/)) |
|---|---|
| Screenshot 2021-01-12 at 15 51 42 | Screenshot 2021-01-12 at 15 51 55 |

Inverse colours haven't changed:
Screenshot 2021-01-12 at 15 51 42

## Checklist
- [x] I have considered the impact of this change and added a [Change Classification](
https://transferwise.atlassian.net/wiki/spaces/EKB/pages/1401189673/Change+Classifications+and+Expectations) Label (`change:standard`, `change:impactful`, `change:emergency`)
- [x] Change meets or does not compromise the [Baseline Security Requirements](https://transferwise.atlassian.net/wiki/spaces/EKB/pages/434929973/Baseline+Security+Requirements)

### CODE REVIEW

File: src/PublicNavigation/Navigation/Menu/Menu.less [hunk 1]
```diff
@@ -112,8 +112,11 @@
           .tw-public-navigation-item-content {
             &__text {
               font-size: @font-size-medium;
-              color: @brand-white;
               max-width: 100vw;
+
+              .navbar--inverse & {
```
Comment (KarlEST) [inline src/PublicNavigation/Navigation/Menu/Menu.less:117]: Do we need to use the `&` in the end or can we use something else that's easier to understand?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+# v8.1.5
+
+## Bugfixes
+
+• Fix visual styles for the white-background (non-inverse) nav
```
Comment (karanjthakkar) [inline CHANGELOG.md:5]:
```suggestion
• Fix visual styles for the white-background (non-inverse) nav

```

File: src/PublicNavigation/Navigation/Navigation.less [hunk 3]
```diff
@@ -371,8 +373,11 @@
 
           strong {
             font-weight: 100;
-            color: @brand-white;
             line-height: 24px;
+
+            .navbar--inverse& {
```
Comment (brettyukich) [inline src/PublicNavigation/Navigation/Navigation.less:378]: Should there be a space between `.nav-bar--inverse` and the `&` here?

Comment (nabaroa) [inline src/PublicNavigation/Navigation/Navigation.less:378]: Space missing?

File: src/PublicNavigation/Navigation/Menu/Menu.less (hunk 1 above)
Comment (danlaush) [reply src/PublicNavigation/Navigation/Menu/Menu.less:117]: I could recreate the selector in a separate tree. I like the simplicity of how this is written and eliminates duplication, but I know how much of a preprocessor-y syntax it is.

**Review by brettyukich — APPROVED** [review]

File: src/PublicNavigation/Navigation/Menu/Menu.less (hunk 1 above)
Comment (danlaush) [reply src/PublicNavigation/Navigation/Menu/Menu.less:117]: image

**Review by nabaroa — APPROVED** [review]

File: src/PublicNavigation/Navigation/Navigation.less (hunk 3 above)
Comment (danlaush) [reply src/PublicNavigation/Navigation/Navigation.less:378]: This one is on purpose - the highest selector is `.navbar--redesign` which is on the same element as `.navbar--inverse`. Without a space, it compiles to `.navbar--inverse.navbar--redesign` which is needed to make this selector work. 

@KarlEST this feels even more hacky than the line you commented on 🙈

**Review by karanjthakkar — APPROVED** [review]
LGTM. Don't wanna block this but same feedback as Karl. The `&` selector is a bit strange to grok if you aren't used to it. If it's easy to copy paste the hierarchy and style using that, it would be consistent with how I've done most of the overriding, not that it's better in anyway.

File: src/PublicNavigation/Navigation/Navigation.less (hunk 3 above)
Comment (karanjthakkar) [reply src/PublicNavigation/Navigation/Navigation.less:378]: Oh boy. i thought the missing space was just a mistake 😅

**Comment by danlaush** [comment]
Thanks for the feedback everyone - have gone the more explicit route 👍  the CSS in there has gotten convoluted over the years, and starting from the bootstrap base. I wonder, is now a good time to re-arch the styles? Hopefully we'll be using this design for a while 😅

**Pull request merged**